app/processors/shopping/cart_purchase_processor.rb

```diff
@@ -14,6 +14,10 @@ def create_resource
         raise Shopping::NotAuthorizedError.new('not authorized', resource_klass: Shopping::CartPurchase)
       end
 
+      if (other_cart = Cart.where(origin: cart.origin, failed_at: nil, purchased_at: nil, user_id: cart.user_id).where.not(id: cart.id).first) && other_cart.cart_purchases.incomplete.exists?
```

Comment:
possible to declare `other_cart` above the `if` for readability and then just checkout `other_cart && ther_cart.cart_purchases.incomplete.exists?` ?

Reply:
For sure